comp/core/workloadmeta/collectors/internal/ecs/v4parser.go

```diff
@@ -0,0 +1,126 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+
+package ecs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta/collectors/util"
+	v1 "github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/v1"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/v3or4"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// parseV4Tasks queries the v4 task endpoint for each task, parses them and stores them in the store.
+func (c *collector) parseTasksFromV4Endpoint(ctx context.Context) ([]workloadmeta.CollectorEvent, error) {
+	tasks, err := c.metaV1.GetTasks(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	events := []workloadmeta.CollectorEvent{}
+	seen := make(map[workloadmeta.EntityID]struct{})
+
+	taskWorker := newWorker(c.taskRateRPS, c.taskRateBurst, c.taskCache, c.getTaskWithTagsFromV4Endpoint)
```

Comment:
Would it make sense to specify the type as part of `newWorker`. That way is clearer that `newWorker` is a functio using generics. `newWorker[v3or4.Task](..)`

Reply:
Also, it seems that we are currently using for v4. Are we planning to use it for other types? Do we need generics?

Reply:
I addressed you feedback about `newWorker[v3or4.Task](..)`

Yes, if later v5 is available then we will use v5.